.github/actions/buildkite/action.yml

```diff
@@ -0,0 +1,141 @@
+---
+name: Buildkite Pipeline GitHub Action
+description: A GitHub Action for triggering a build on a Buildkite pipeline.
+inputs:
+  vaultUrl:
+    description: 'Vault URL'
+    required: true
+  vaultRoleId:
+    description: 'Vault role ID'
+    required: true
+  vaultSecretId:
+    description: 'Vault secret ID'
+    required: true
+  secret:
+    description: 'Vault secret with the token field.'
+    default: secret/observability-team/ci/buildkite-automation
+    required: false
+  org:
+    description: 'BuildKite org to interact with.'
+    default: elastic
+    required: false
+  pipeline:
+    description: 'BuildKite pipeline to interact with.'
+    required: true
+  waitFor:
+    description: 'Whether to wait for the build to finish.'
+    default: false
+    required: false
+  printBuildLogs:
+    description: 'Whether to wait for the build to finish.'
+    default: false
+    required: false
+  buildEnvVars:
+    description: 'Additional environment variables to set on the build.'
+    required: false
+
+runs:
+  using: "composite"
+  steps:
+      - uses: hashicorp/vault-action@v2.4.2
+        with:
+          url: ${{ inputs.vaultUrl }}
+          roleId: ${{ inputs.vaultRoleId }}
+          secretId: ${{ inputs.vaultSecretId }}
+          method: approle
+          secrets: |
+            ${{ inputs.secret }} token | BUILDKITE_API_ACCESS_TOKEN
+
+      - name: Trigger Buildkite pipeline
+        shell: bash
+        run: |
```

Comment:
you can call a script in the same folder, maybe is better to put the script in a separate file

```yaml
    - run: echo "${{ github.action_path }}" >> $GITHUB_PATH
      shell: bash
    - run: action-script.sh
      shell: bash
```

Reply:
Interesting! Let me try it

Reply:
See https://github.com/elastic/apm-pipeline-library/pull/2033/commits/297cf4dbf900ac68189cc3220e5203eabe9701db